tasks/__init__.py

```diff
@@ -43,109 +43,76 @@ def requirements(ctx, dev=False):
-    test_module(ctx)
+    test_module(ctx, module=['-m','smoke_test'])
 
 @task
-def test_travis_edge(ctx, numprocesses=None):
-    """
-    Run tests on the latest Edge
-    """
-    flake(ctx)
-    print('Testing modules in "{}" in Edge'.format('tests'))
-    test_module(ctx)
+def test_travis_part_one(ctx):
+    """Run first group of tests on the browser defined by TEST_BUILD."""
+    all_test_files = _get_test_file_list()
+    midpoint = len(all_test_files) // 2
+    file_list = all_test_files[:midpoint]
+    test_travis_with_retries(ctx, 'part one', file_list)
 
 @task
-def test_travis_firefox(ctx, numprocesses=None):
-    """
-    Run tests on the latest Firefox
-    """
-    flake(ctx)
-    print('Testing modules in "{}" in Firefox'.format('tests'))
-    test_module(ctx)
+def test_travis_part_two(ctx):
+    """Run second group of tests on the browser defined by TEST_BUILD."""
+    all_test_files = _get_test_file_list()
+    midpoint = len(all_test_files) // 2
+    file_list = all_test_files[midpoint:]
+    test_travis_with_retries(ctx, 'part two', file_list)
+
+def _get_test_file_list():
+    all_test_files = glob.glob('tests/test_*.py')
+    all_test_files.sort()
+    return all_test_files
 
 @task
-def test_travis_android(ctx, numprocesses=None):
-    """
-    Run tests on Android 7.0, Samsung Galaxy S8
-    """
+def test_travis_with_retries(ctx, partition_name, file_list):
+    """Run group of tests on the browser defined by TEST_BUILD."""
     flake(ctx)
-    print('Testing modules in "{}" in android'.format('tests'))
-    test_module(ctx)
 
-@task
-def test_travis_ios(ctx, numprocesses=None):
-    """
-    Run tests on ios 10.0, iPhone 7
-    """
-    flake(ctx)
-    print('Testing modules in "{}" on ios'.format('tests'))
-    test_module(ctx)
+    print('Testing {} modules in "/tests/" in {}'.format(partition_name, os.environ['TEST_BUILD']))
+    print('File list for {} is: {}'.format(partition_name, file_list))
+    retcode = test_module_wo_exit(ctx, params=file_list)
 
-@task
-def test_travis_on_prod(ctx, numprocesses=None):
-    """
-    Runs targeted prod smoke tests on the latest Chrome
-    """
-    flake(ctx)
-    print('Testing modules in "{}" in Chrome'.format('tests'))
-    test_module(ctx, module=['-m','smoke_test'])
+    if retcode != 1:
+        sys.exit(retcode)
 
-@task
-def test_travis_failures_only_chrome(ctx, numprocesses=None):
-    """
-    Run tests on the latest Chrome
-    """
-    print('Testing modules in "{}" in Chrome'.format('tests'))
-    test_module(ctx, params=['--last-failed', '--last-failed-no-failures', 'none'])
+    file_list = ['--last-failed', '--last-failed-no-failures', 'none'] + file_list
 
-@task
-def test_travis_failures_only_edge(ctx, numprocesses=None):
-    """
-    Run tests on the latest Edge
-    """
-    print('Testing modules in "{}" in Edge'.format('tests'))
-    test_module(ctx, params=['--last-failed', '--last-failed-no-failures', 'none'])
+    for i in range(1, 4):
```

Comment:
Nitpicking ahead ⚠️ 

* I don't remember why we came up with the `4` as the magic number. So adding a comment?
* A further improvement is to make `4` configureable, either via pytest args or selenium settings, but this is probably overkill